extensions/amp-carousel/0.2/slide-manager.js

```diff
@@ -0,0 +1,274 @@
+/* eslint-disable no-unused-vars */
+
+import {
+  Alignment,
+  Axis,
+} from './dimensions.js';
+
+export class SlideManager {
```

Comment:
> Since this is a review for a shell I'm going to be nitpicky with design.

That was a good part of the intent, so if there is anything that stands out from a code organization perspective, better to get that sorted out up front.

> It's unclear to me why SlideManager exists. It seems like its responsibilities are not clearly defined. Why should a thing that manages slides update the UI? How do slides get... managed?

They get moved around (using translate) as you slide. The UI update is just to move the UI into the correct state after you change a configuration option (e.g. show 3 slides at a time instead of 1). I originally had the snapping logic was outside of SlideManager, but I had to fold it back in. I can move the logic up into carousel. I may refactor this sometime in the future if I can split out responsibilities in a way that makes sense (and I have some tests).

Reply:
Cool, it makes sense to keep them there but I would be weary of overloading `SlideManager` more.